libs/gpg45-evaluator/src/main/java/uk/gov/di/ipv/core/library/gpg45/validators/Gpg45IdentityCheckValidator.java

```diff
@@ -0,0 +1,56 @@
+package uk.gov.di.ipv.core.library.gpg45.validators;
+
+import uk.gov.di.ipv.core.library.annotations.ExcludeFromGeneratedCoverageReport;
+import uk.gov.di.ipv.core.library.domain.Cri;
+import uk.gov.di.model.IdentityCheck;
+
+import static software.amazon.awssdk.utils.CollectionUtils.isNullOrEmpty;
+import static uk.gov.di.ipv.core.library.gpg45.Gpg45ProfileEvaluator.getVerificationScore;
+
+public class Gpg45IdentityCheckValidator {
+    @ExcludeFromGeneratedCoverageReport
+    private Gpg45IdentityCheckValidator() {
+        throw new IllegalStateException("Utility class");
+    }
+
+    public static boolean isSuccessful(IdentityCheck identityCheck, Cri cri) {
+        return switch (cri) {
+            case BAV, DRIVING_LICENCE, PASSPORT -> isEvidenceSuccessful(identityCheck);
+            case EXPERIAN_FRAUD -> isFraudCheckSuccessful(identityCheck);
+            case DWP_KBV, EXPERIAN_KBV, HMRC_KBV -> isVerificationSuccessful(identityCheck);
+            case DCMAW, DCMAW_ASYNC, F2F, HMRC_MIGRATION -> isEvidenceSuccessful(identityCheck)
+                    && isVerificationSuccessful(identityCheck);
+            case NINO -> isNinoSuccessful(identityCheck);
+            case ADDRESS,
+                    CIMIT,
+                    CLAIMED_IDENTITY,
+                    TICF -> false; // These CRIs should not perform identity checks
```

Comment:
They're not "unsuccessful". Makes we wonder if the method should be called something more specific to GPG45? `isGpg45Success`? Or something.

Reply:
Yeah, perhaps this should just be an error instead - if one of these _is_ presenting an IdentityCheck, that's an issue!

(I just got a bit nervous about tightening validation...)

Reply:
I think I'd prefer to:
- If we want to make it an error, validate in the `VerifiableCredentialValidator` (SPOT-style)
- Leave this as a non-throwing check, but with a warning log

Reply:
Yeah. that seems fair